src/pages/index.js

```diff
@@ -45,61 +68,177 @@ popupPreviewImage.setEventListeners();
+popupProfileImageForm.setEventListeners();
 
-//Section Class
-const section = new Section(
-  { items: initialCards, renderer: renderCard },
-  cardListEl
+//PopupWithDelete Class
+const popupWithDeleteCard = new PopupWithDelete(
+  deleteCardModal,
+  loadingButtonText
 );
-section.renderItems();
+popupWithDeleteCard.setEventListeners();
 
 //UserInfo Class
-const userClass = new UserInfo({
+const userInfo = new UserInfo({
   nameContainer: profileName,
   descriptionContainer: profileDescription,
+  profilePic,
+});
+
+//Api Class
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1755563a-df02-4ecd-aa91-5c5958bedadf",
+    "Content-Type": "application/json",
+  },
 });
 
+//Section Class
+const section = new Section({ renderer: renderCard }, cardListEl);
+
+api
+  .getInitialCards()
+  .then((cardItems) => {
+    section.renderItems(cardItems);
+  })
+  .catch((err) => console.log(err));
+
+api
+  .getProfileInfo()
+  .then((res) => {
+    // profilePic.src = res.avatar;
+    userInfo.setAvatar(res);
+
+    return userInfo.setUserInfo({
+      nameInput: res.name,
+      descriptionInput: res.about,
+    });
+  })
+  .catch((err) => console.log(err));
+
+// initialCards.forEach((card) => {
+//   api.addCard(card);
+// });
+
 //
 //Functions
 function fillProfileInputs() {
-  const inputs = userClass.getUserInfo();
+  const inputs = userInfo.getUserInfo();
 
   profileNameInput.value = inputs.profileName;
   profileDescriptionInput.value = inputs.profileDescription;
 }
 
 function handleProfileFormSubmit(data) {
-  userClass.setUserInfo({
-    nameInput: data.title,
-    descriptionInput: data.description,
-  });
-
-  popupProfileForm.close();
+  api
+    .updateProfileInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then(() => {
+      popupProfileForm.showLoading();
+      userInfo.setUserInfo({
+        nameInput: data.title,
+        descriptionInput: data.description,
+      });
+      popupProfileForm.close();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      popupProfileForm.hideLoading();
+    });
 }
 
 function handleCardFormSubmit(data) {
   const name = data.title;
   const link = data.description;
-  renderCard({ name, link });
-  popupCardForm.close();
-  cardEditForm.reset();
-  addFormValidator.toggleButtonState();
+  api
+    .addCard({ name: name, link: link })
+    .then(() => {
+      popupCardForm.showLoading();
+      renderCard({ name, link });
+      popupCardForm.close();
+      cardEditForm.reset();
+      addFormValidator.toggleButtonState();
+    })
+    .catch((err) => console.log(err))
+    .finally(() => {
+      popupCardForm.hideLoading();
+    });
+  // renderCard({ name, link });
+  // popupCardForm.close();
+  // cardEditForm.reset();
+  // addFormValidator.toggleButtonState();
+}
+
+function handleProfileImageFormSubmit(data) {
+  profilePic.src = profilePicModalInput.value;
```

Comment:
Needs correction:

1) you should use setAvatar method for updating the picture
2) you should update the picture only after the successful api response as it should not be changed in the case of an error